docker/pre_push_hook.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+#
+# Copyright 2024 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# For this script to work properly, you need to have scripts/pre_push_hook.py
+# This scripts helps running pre-push hook inside docker container by,
+# working as middle layer beteen hook and pre_push_hook.py in docker
+# If you can't understand whats going on here, please visit
+# https://docs.google.com/document/d/1whHSUEchpmnqxSsRvHU_WYbBzeqVMh7vrjpuA8FMnAM/edit?usp=sharing
+#
+# To install hook (which runs inside docker container), run this script
+# command from oppia root directory with --install argument
+#
+# Currently it only works for Unix systems, on windows it will create symlink but won't have any effect
+
+DEV_CONTAINER="dev-server"
+DOCKER_EXEC_COMMAND="docker compose exec -T $DEV_CONTAINER "
+
+
+# Location of git hooks directory
+HOOKS_DIR=".git/hooks"
+
+# Path for symbolic links to hook files
+PRE_PUSH_SYMLINK="$HOOKS_DIR/pre-push"
+PYTHON_PRE_PUSH_SYMLINK="$HOOKS_DIR/pre-push-python"
+
+# Path for hook files
+PRE_PUSH_FILE="docker/pre_push_hook.sh"
+PYTHON_PRE_PUSH_FILE="scripts/pre_push_hook.py"
+
+# Install pre-push hook
+install_hook() {
+    # Create symlinks for pre-push hook files
+    for file in "$PRE_PUSH_SYMLINK" "$PYTHON_PRE_PUSH_SYMLINK"; do
```

Comment:
We are not expecting to run pre-push checks this way in the docker setup. Instead we are planning to run separate containers for each check i.e, fronted tests on fronted service, separate container for lint checks made on top of the Backend image. 

I'm fine with keeping it this way for now as it's blocking developers, can you please create a seperate issue tracking this task? (I can add details on the issue as needed.)

Reply:
Created #19915

Reply:
Also Sandeep, do you mean, we want the pre-push hook to be rewritten, so we can take advantage of running multiple containers simultaneously for different tests? Also new hook would be written in shell script.

Reply:
It would be less of a complex script or more of a set of 3 commands to run the different containers (So this can be used in any OS) and now these different containers will run the required tests.

Reply:
@jayam04 We will proceed with this approach for now and open and issue top address this as part of a larger project.